great_expectations/experimental/datasources/spark_s3_datasource.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import TYPE_CHECKING, Any, Dict, Optional, Union
+
+import pydantic
+from typing_extensions import Literal
+
+from great_expectations.core.util import S3Url
+from great_expectations.experimental.datasources import _SparkFilePathDatasource
+from great_expectations.experimental.datasources.data_asset.data_connector import (
+    DataConnector,
+    S3DataConnector,
+)
+from great_expectations.experimental.datasources.interfaces import TestConnectionError
+from great_expectations.experimental.datasources.spark_file_path_datasource import (
+    CSVAsset,
+)
+
+if TYPE_CHECKING:
+    from botocore.client import BaseClient
+
+    from great_expectations.experimental.datasources.interfaces import (
+        BatchSorter,
+        BatchSortersDefinition,
+    )
+
+
+logger = logging.getLogger(__name__)
+
+
+try:
+    import boto3
+except ImportError:
+    logger.debug("Unable to load boto3; install optional boto3 dependency for support.")
+    boto3 = None
+
+
+class SparkS3Datasource(_SparkFilePathDatasource):
+    # instance attributes
+    type: Literal["spark_s3"] = "spark_s3"
+
+    # S3 specific attributes
+    bucket: str
+    boto3_options: Dict[str, Any] = {}
+
+    _s3_client: Optional[BaseClient] = pydantic.PrivateAttr()
+
+    def __init__(self, **data):
+        super().__init__(**data)
+
+        try:
+            self._s3_client = boto3.client("s3", **self.boto3_options)
+        except (TypeError, AttributeError):
+            self._s3_client = None
+
+    def test_connection(self, test_assets: bool = True) -> None:
+        """Test the connection for the SparkS3Datasource.
+
+        Args:
+            test_assets: If assets have been passed to the SparkS3Datasource, whether to test them as well.
+
+        Raises:
+            TestConnectionError: If the connection test fails.
+        """
+        if self._s3_client is None:
+            raise TestConnectionError(
+                "Unable to load boto3 (it is required for SparkS3Datasource)."
+            )
+
+        if self.assets and test_assets:
+            for asset in self.assets.values():
+                asset.test_connection()
+
+    def add_csv_asset(
+        self,
+        name: str,
+        batching_regex: Union[re.Pattern, str],
```

Comment:
This should be optional, like it is here: https://github.com/great-expectations/great_expectations/blob/b9eeeb14f47cf754dabd4bb2cbcc3aa75aeb66d4/great_expectations/experimental/datasources/pandas_filesystem_datasource.py#L62

Reply:
@NathanFarmer Done!  Thanks!